### PR TITLE
Implement stepping through progress bar

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/view/ProgressBar.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    case QuestionState.WRONG:
+      return "#c11818";
+    case QuestionState.CORRECT:
+      return "#2e9937";
+  }
+};
+
+const getFillColor = (state: QuestionState) => {
+  switch (state) {
+    case QuestionState.UNANSWERED:
```
Comment (JoshAdamPowell) [inline src/view/ProgressBar.tsx]: I asked Olly & Carly to put the colours into a dedicated file, they'll be merging their work to main soon, can you rebase once they have and also put these colours in that file?

File: src/view/ProgressBar.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      return "#eeeded";
+    case QuestionState.WRONG:
+      return "#de2d2c";
+    case QuestionState.CORRECT:
+      return "#4caf4c";
+  }
+};
+
+const ProgressDot = styled.div`
```
Comment (JoshAdamPowell) [inline src/view/ProgressBar.tsx]: Also asked them to put the styled components in a separate file. Check the climate change app to see how we have it there but it's a folder named Component that contains both Component.tsx and styled.ts which contains all the styles.

File: src/view/QuizView.tsx [hunk 3]
```diff
@@ -1,10 +1,41 @@
...
+      : QuestionState.WRONG;
+    setQuestionStates(newQuestionStates);
+    setCurrentIndex(currentIndex + 1);
+  };
+
+  return (
+    <>
+      <ProgressBar questionStates={questionStates} />
+      <button onClick={() => updateResult(Math.random() < 0.5)}>
+        Next Question{" "}
```
Comment (JoshAdamPowell) [inline src/view/QuizView.tsx]: not your fault but prettier likes to add these weird `{" "}` to jsx if you leave a space at the end of the line. I dont like it.

**Review by JoshAdamPowell — CHANGES_REQUESTED** [review]

File: src/view/QuizView.tsx (hunk 3 above)
Comment (ghost) [reply src/view/QuizView.tsx]: Good spot, have removed. This is a placeholder component anyway

File: src/view/ProgressBar.tsx (hunk 2 above)
Comment (ghost) [reply src/view/ProgressBar.tsx]: done

File: src/view/ProgressBar.tsx (hunk 1 above)
Comment (ghost) [reply src/view/ProgressBar.tsx]: done

**Review by JoshAdamPowell — APPROVED** [review]

**Pull request merged**